### DISCUSSION TITLE
The next big developments for the platform (V2)

### DISCUSSION THREAD

**Discussion by davehakkens** [ONEARMY/community-platform#1108] [General]
Some of you might have already seen [this document](https://docs.google.com/document/d/1hyWc2a9cn04B107UzUKd4-12T5v0VrlWmYTa8KZnYhQ/edit) floating around somewhere. Its a Gdoc that shows the 3 main developments for V2  

- ⭐ Useful button
- 🔬 Research module
- 💬 Comments  

The goal of Version 2 is to have all our modules online for uploading content. The research module was the main missing element up until now. Plus V2 will bring community interaction with Comments and “Liking” content on the platform. 

**Useful button**
   The useful button was taken on by @tudi2d 
   
      **Research module**
   Next we want to work on the Research module. @chrismclarke already did a lot of the groundwork in the backend. You can see some of it on [the devsite](https://dev.onearmy.world/academy)    And a [basic issue](https://github.com/ONEARMY/community-platform/issues/1080) is created, the main work to be done for this is the frontend. Perhaps interesting for @tudi2d or @danitrod?
   
   **Comments**
   We have a plan (layout) for this in the document. But it is a complex issue, especially if you want to have it working smoothly in all the details. (Not even talking about notifications and spam) Perhaps anyone has experience or suggestions for this module? 
   
 This topic is mainly to have a place to chat about V2. Throw all your comments, feedback and suggestions in here

**Comment by danitrod** [comment]
I'd be happy with either continuing on the work on research module or starting to work on the comments! I guess for now it's better to finish up research, so I'll start looking into how to help this week

**Reply by chrismclarke** [reply]
Great, I'll get in touch on the open PRs I can see and specific git issues to provide more info there

**Comment by tudi2d** [comment]
I like the idea of working with GitHub discussions here! 👍 

An idea for the comments: For the first version we could make it, that one user can only add one comment (with a limited length) & no replies are possible. I understand the value of the comments mostly in giving feedback, adding similar resources & connecting. This would be possible & at the same time spam & discussions were avoided. For deeper discussions there is always Discord or Mail.

**Reply by davehakkens** [reply]
Makes sense.  I would be up for taking this approach and release step by step.

**Reply by chrismclarke** [reply]
I really like that idea, partly because it would be easier for a first iteration from a technical side, but also if it encourages more of "this is constructive feedback" rather than "let's start a discussion". Maybe we could also perhaps allow the comments within a specific block/step of the text, to keep it focused (a bit like how ifixit let's users elaborate on things they tried during specific stages from the instructions) - and users who want to comment more can add more individual section comments (as opposed to trying to talk about everything in a single thread)

**Reply by chrismclarke** [reply]
Just trying to remember, @davehakkens is there already a google doc outlining the comments module? If so maybe we can share a link here to invite for comments before creating a couple GitHub issues?

**Reply by tudi2d** [reply]
>is there already a google doc outlining the comments module?

@chrismclarke It's linked in the original post of this discussion & the part on comments starts on page 8 :)

**Comment by davehakkens** [comment]
All released :)
